// Copyright 2026 Poolside, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package poolside.bridge;

option go_package = "github.com/poolsideai/forge/pkg/bridge/proto";

// Bridge service for managing agent operations
service BridgeSidecarService {
  // Start an agent with a given prompt
  rpc StartAgent (StartAgentRequest) returns (StartAgentResponse);
}

message RunDetail {
  string agent_name = 1;
  string session_id = 2;
}

message ContinueFrom {
  RunDetail previous_run_detail = 1;

  message NoCompactionStrategy {}
  message CompactionStrategy {}
  oneof compaction_strategy {
    NoCompactionStrategy continuation = 2;
    CompactionStrategy compaction = 3;
  }
}

message StartAgentRequest {
  string prompt = 1;
  string agent_name = 2;
  string directory = 3;
  ContinueFrom continue_from = 4;
}

message StartAgentResponse {
  // Unique identifier for the agent
  RunDetail run_detail = 1;

  // Result/exit status of the agent
  string exit_result = 3;
}
